examples/data_ops/1151_digits.py

```diff
@@ -0,0 +1,165 @@
+# in a declarative way using the apply method.
+
+from sklearn.svm import SVC
+
+model = SVC()
+predictions = X.skb.apply(model, y=y)
+
+# %%
+# After defining the computational plan with apply, we create a learner object
+# and evaluate it using cross-validation. This gives us a baseline performance
+# metric for our SVM classifier.
+
+learner = predictions.skb.make_learner(fitted=True)
+
+cv_results = skrub.cross_validate(learner, environment=predictions.skb.get_data())
+print(f"Baseline SVM - Mean CV accuracy: {cv_results['test_score'].mean():.4f}")
+
+# %%
+# Hyperparameter Tuning with Multiple Models
+# ============================================
+#
+# To improve performance, we'll now explore multiple algorithms and their
+# hyperparameters using skrub's hyperparameter tuning capabilities.
+# The choose_from and choose_* functions define a search space over which
+# we can perform randomized search.
+
+from sklearn.ensemble import HistGradientBoostingClassifier, RandomForestClassifier
+
+# %%
+# Define a search space with three different algorithms:
+# 1. **SVM with C and gamma tuning**: Regularization and kernel parameters
+# 2. **Random Forest**: Number of trees and tree depth
+# 3. **Histogram Gradient Boosting**: Iterations, depth, and learning rate
+
+model = skrub.choose_from(
+    {
+        "SVC": SVC(
+            C=skrub.choose_float(0.1, 10.0, name="C", log=True),
+            gamma=skrub.choose_float(0.001, 1.0, name="gamma", log=True),
+        ),
+        "RandomForest": RandomForestClassifier(
+            n_estimators=skrub.choose_int(10, 200, name="n_estimators"),
+            max_depth=skrub.choose_int(5, 50, name="max_depth_rf"),
+        ),
+        "HistGradientBoosting": HistGradientBoostingClassifier(
+            max_iter=skrub.choose_int(50, 200, name="max_iter"),
+            max_depth=skrub.choose_int(3, 20, name="max_depth_hgb"),
+            learning_rate=skrub.choose_float(0.01, 0.5, log=True, name="learning_rate"),
+        ),
+    }
+)
+
+predictions = X.skb.apply(model, y=y)
+
+# %%
+# Running Randomized Search
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We perform a randomized search over the hyperparameter space. This involves:
+# - Randomly sampling 50 different hyperparameter combinations
+# - Using 5-fold cross-validation to evaluate each combination
+# - Running evaluations in parallel (n_jobs=-1)
+
+search = predictions.skb.make_randomized_search(
+    n_iter=50, cv=5, random_state=42, n_jobs=-1, fitted=True
+)
+
+# %%
+# Visualizing Search Results
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Let's plot the search results to see how different hyperparameter
+# combinations affected performance:
+
+search.plot_results()
+
+# %%
+# Comparing Results
+# ~~~~~~~~~~~~~~~~~
+#
+# Finally, we compare the performance of our baseline model with the
+# best model found during hyperparameter tuning:
+
+best_learner = search.best_learner_
+cv_results_search = skrub.cross_validate(
```

Comment:
technically we are overfitting the hyperparameters. maybe we should split away a test set at the beginning that we don't use for hyperparam search?